long polling support

BarAgam opened the pull request:
long polling support

barmako committed:
long polling support

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GGMessaging/pull/2" target="_blank"><img src="https://app.review.ninja/54319889/pull/2/badge" alt="ReviewNinja"/>

teamagam3 commented on routes/common.js:
```diff
@@ -0,0 +1,86 @@
+            //forward to error handling
+            next(err);
+        }
+
```
Comment is missing here

barmako committed:
long polling support version 2

barmako committed:
long polling support.
Returning appropriate status code on timeout events.
Removes listener of waiting request handlers on timeout

teamagam1 commented on app.js:
```diff
@@ -34,7 +36,20 @@ app.use(bodyParser.urlencoded({extended: false}));
+app.use('/long/messages', longPollingMessagesRoute);
+
+//timeout catcher
+app.use(function (req, res, next) {
```
who is going to first? route to /long/messages or timeout?
does the order in app.js is important?

BarAgam replied:
The order is given by the order the order of the app.use methods. that means a request will first try and match /message and /long/messages route and afterwards (if a response is not already sent) it'll be forwarded to the timeout catcher and the error handler

teamagam1 commented on EventEmitters/mongoEventsEmitter.js:
```diff
@@ -0,0 +1,12 @@
+const EventEmitter = require('events');
+const util = require('util');
+
+function MessageCollectionEmitter() {
```
documentation.
who uses those events and what for?

teamagam1 commented on routes/long_polling_messages.js:
```diff
@@ -0,0 +1,65 @@
+
+    common.getMessagesFromDate(req, function (err) {
+        next(err);
+    }, function (messages) {
```
this is the same code as above.
can't you merge those two to one method?

barmako committed:
increased K (result limit per request) from 3 to 50

teamagam1 commented on routes/common.js:
```diff
@@ -0,0 +1,76 @@
+var config = require('../config/config');
+
+
+function getAllMessages(onFailure, onSuccess) {
```
documentation.
difference between those two methods

teamagam1 commented:
when i used long pulling i got this message:
timeout reached. removing listener [this is OK]
Error: Can't set headers after they are sent.
.
.
.
[stack trace]
!fix
